src/statemachine.rs

```diff
@@ -412,38 +415,37 @@ impl StateMachineCtap2 {
         }
     }
 
-    fn determine_pin_auth<T: PinAuthCommand, U>(
+    fn determine_pin_auth<T: PinUvAuthCommand + Request<V>, U, V>(
         cmd: &mut T,
         dev: &mut Device,
+        skip_uv: bool,
         status: &Sender<StatusUpdate>,
         callback: &StateCallback<crate::Result<U>>,
-    ) -> Result<(), ()> {
-        loop {
-            match cmd.determine_pin_auth(dev) {
-                Ok(_) => {
-                    break;
+    ) -> Result<PinUvAuthResult, PinUvAuthError> {
+        let mut pin_uv_auth_result = PinUvAuthResult::RequestIsCtap1;
+        if cmd.is_ctap2_request() {
```

Comment:
nit: an early return here would be easier to read
```
if !cmd.is_ctap2_request() {
    return Ok(PinUvAuthResult::RequestIsCtap1);
}
```
(and in the `Ok(r)` case below)

Reply:
Done